### ISSUE TITLE
Media players receive focus but show no visible focus state

### ISSUE THREAD

**Issue by karlgroves** [WordPress/gutenberg#15312]
Media players receive focus but show no visible focus state

-   **Severity:** High
-   **Affected Populations:**
    -   Low-Vision
    -   Motor Impaired
    -   Cognitively Impaired
-   **Platform(s):**
    -   All / Universal
-   **Components affected:**
    -   Output Markup

#### Issue description

Keyboard users who are navigating through a published (or previewed)
page may come across media players implemented with native audio and
video elements. These receive keyboard focus in most or all browsers,
but the page does not visually show it. This results in invisible Tab
stops.

All interactive elements should have a visible and obvious focus
indicator, so that sighted keyboard users have a clear indication of
which element has focus.

##### Issue Code

```
    :focus {


        outline: none;


    }

```

#### Remediation Guidance

Whenever the native focus `outline` is removed from focusable
elements, a replacement visual focus state must be provided, with a
minimum contrast ratio of 3:1.

##### Recommended Code

```
    audio:focus, video:focus {


        outline: 1px solid #0085ba;


    }
```

#### Relevant standards

-   [2.1.1 Keyboard
    (Level A)](https://www.w3.org/TR/WCAG20/#keyboard-operation-keyboard-operable)
-   [2.4.7 Focus Visible (Level
    AA)](https://www.w3.org/TR/WCAG20/#navigation-mechanisms-focus-visible)

**Note:** This issue may be a duplicate with other existing accessibility-related bugs in this project. This issue comes from the Gutenberg accessibility audit, performed by [Tenon](https://www.tenon.io) and funded by [WP Campus](https://wpcampus.org/). This issue is GUT-106 in Tenon's report

**Comment by melchoyce** [comment]
Screenshot from the full report:

![image](https://user-images.githubusercontent.com/2846578/57182743-9d31af80-6e57-11e9-86a4-1bf2849112f4.png)

**Comment by afercia** [comment]
Screenshots show the native HTML audio and video players **in the front end**. While these elements are probably difficult to style, it would be great for Gutenberg to provide some default focus styles to improve accessibility of the produced content.

**Comment by kjellr** [comment]
For what it's worth, I'm seeing the default browser focus style picked up here: 

Chrome: 
<img width="750" alt="Screen Shot 2019-05-07 at 12 20 08 PM" src="https://user-images.githubusercontent.com/1202812/57315861-788d4000-70c2-11e9-8dca-5e50c951bcac.png">

Safari: 
<img width="617" alt="Screen Shot 2019-05-07 at 12 20 28 PM" src="https://user-images.githubusercontent.com/1202812/57315883-8642c580-70c2-11e9-8520-65e8748d75a7.png">

Firefox: 
<img width="711" alt="Screen Shot 2019-05-07 at 12 20 51 PM" src="https://user-images.githubusercontent.com/1202812/57315905-9064c400-70c2-11e9-9efb-98f0e4d1c29e.png">

We could theoretically add a focus style here, but since this is the front end, our focus style may be different than other focus styles enabled in the theme.

**Comment by afercia** [comment]
I guess the missing focus style is on the player controls 🙂Nothing is displayed when tabbing through Play/Pause, slider, volume, etc.

**Comment by kjellr** [comment]
It looks like that's up to the browser. I'm able to select the interior buttons in Safari and Chrome, but not in Firefox. 

_Safari:_ 
<img width="611" alt="Screen Shot 2019-05-07 at 2 44 49 PM" src="https://user-images.githubusercontent.com/1202812/57324913-e6436700-70d6-11e9-8224-4030e84d7de3.png">

_Chrome:_
<img width="723" alt="Screen Shot 2019-05-07 at 2 45 01 PM" src="https://user-images.githubusercontent.com/1202812/57324922-e9d6ee00-70d6-11e9-9dc2-9d963627ea21.png">


If we can confirm that this issue is about the interior buttons, I'm not sure there's anything we can do about it — we can't touch them via CSS unfortunately.

**Comment by afercia** [comment]
Right, in Firefox I see the controls are styled via xul/XML with its own stylesheet:
resource://gre-resources/html.css
chrome://global/content/bindings/videocontrols.xml#videoControls
chrome://global/skin/media/videocontrols.css

Not sure how to change them. 

Maybe start simple with the video/audio elements? 🙂

**Comment by kjellr** [comment]
I don't think we should mess with browser-provided styles within those media players. Even if we can access those styles, we'd need to target each browser specifically and it seems fairly hacky. 

That said: After a bit more digging, I inferred from the screenshot that Tenon tested with Twenty Seventeen. It looks like Twenty Seventeen actually [removes all `:focus` styles by default](https://themes.trac.wordpress.org/browser/twentyseventeen/2.2/style.css#L576), which causes the bug described here.

I'll open a Trac ticket + patch for this, and then link back here. Once that's done, I suggest we close this one as migrated over. Does that seem reasonable to everyone?

**Comment by kjellr** [comment]
I moved this over to trac here (and included a patch to fix it): 

https://core.trac.wordpress.org/ticket/47190

Not sure what happens to the ticket here — do we close it in this case?

**Comment by afercia** [comment]
Yep, if there's really nothing Gutenberg can do please do feel free to close it, thanks!

**Comment by kjellr** [comment]
Great, thanks everyone!
